rpm/packagescanner.go

```diff
@@ -201,26 +204,56 @@ func (ps *Scanner) Scan(ctx context.Context, layer *claircore.Layer) ([]*clairco
 				}()
 				zlog.Debug(ctx).Str("file", f.Name()).Msg("copying sqlite db out of tar")
 				if _, err := io.Copy(f, r); err != nil {
-					return err
+					return nil, fmt.Errorf("rpm: error reading sqlite db: %w", err)
 				}
 				if err := f.Sync(); err != nil {
-					return err
+					return nil, fmt.Errorf("rpm: error reading sqlite db: %w", err)
 				}
-				rpm, err := sqlite.Open(f.Name())
-				defer rpm.Close()
-				infos, err := rpm.Packages(ctx)
+				sdb, err := sqlite.Open(f.Name())
 				if err != nil {
-					return err
+					return nil, fmt.Errorf("rpm: error reading sqlite db: %w", err)
 				}
-				ps, err := packagesFromInfos(ctx, db.String(), infos)
+				defer sdb.Close()
+				nat = sdb
+			case kindNDB:
+				f, err := sys.Open(path.Join(db.Path, `Packages.db`))
 				if err != nil {
-					return err
+					return nil, fmt.Errorf("rpm: error reading ndb db: %w", err)
 				}
-				pkgs = append(pkgs, ps...)
-				return nil
-			}(); err != nil {
-				return nil, fmt.Errorf("rpm: error reading sqlite db: %w", err)
+				defer f.Close()
+				r, ok := f.(io.ReaderAt)
```

Comment:
I'm a little confused here, will we ever get a `ReaderAt` from tarfs.Open() or is this in case tarfs.File fulfills more interfaces in the future?

Reply:
The latter -- no point in spooling out to a temp file if/when my skunkworks changes that would allow for implementing io.ReaderAt land.